From b190963d5d1b03db573f8d9a305c8bb2fa4f353c Mon Sep 17 00:00:00 2001
From: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Date: Tue, 30 Dec 2025 02:21:29 +0200
Subject: [PATCH] meson: allow overriding systemd location

Allow users to specify and/or override systemd unit location instead of
depending on systemd presense at compile time.

Signed-off-by: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Upstream-Status: Inappropriate [qrtr dropped qrtr-ns support]
---
 meson.build | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/meson.build b/meson.build
index 792e8aa75fa2..2d046c2d3f23 100644
--- a/meson.build
+++ b/meson.build
@@ -12,14 +12,15 @@ project('qrtr',
 prefix = get_option('prefix')
 with_qrtr_ns = get_option('qrtr-ns')
 
-install_systemd_unit = get_option('systemd-service')
-systemd = dependency('systemd', required : install_systemd_unit)
-if systemd.found()
-        systemd_system_unit_dir = get_option('systemd-unit-prefix')
-        if systemd_system_unit_dir == ''
+install_systemd_unit = get_option('systemd-service').allowed()
+if with_qrtr_ns.allowed() and install_systemd_unit
+        systemd = dependency('systemd', required : false)
+        if get_option('systemd-unit-prefix') != ''
+                systemd_system_unit_dir = get_option('systemd-unit-prefix')
+        elif systemd.found()
                 systemd_system_unit_dir = systemd.get_variable(
                         pkgconfig : 'systemdsystemunitdir',
-                pkgconfig_define: ['prefix', prefix])
+                        pkgconfig_define: ['prefix', prefix])
         else
                 message('Could not resolve systemd dependencies, skipping unit file')
                 install_systemd_unit = false
@@ -31,7 +32,7 @@ subdir('lib')
 subdir('include')
 subdir('src')
 
-if systemd.found() and with_qrtr_ns.enabled()
+if with_qrtr_ns.allowed() and install_systemd_unit
         systemd_unit_conf = configuration_data()
         systemd_unit_conf.set('prefix', prefix)
         configure_file(
-- 
2.51.0

